pkg/kubelet/cm/device_plugin_handler.go

```diff
@@ -91,6 +92,8 @@ type DevicePluginHandler interface {
 	// the input container, issues an Allocate rpc request for each of such
 	// resources, and returns their AllocateResponses on success.
 	Allocate(pod *v1.Pod, container *v1.Container, activePods []*v1.Pod) ([]*pluginapi.AllocateResponse, error)
+	// Admit determine if device resources requests by the pod is satisfied
+	Admit(attrs *lifecycle.PodAdmitAttributes) lifecycle.PodAdmitResult
```

Comment:
Instead of calling it Admit, maybe call it GetResources. This function can be called from cm.GetResources() and basically, it should just get device runtime information from the cached state. The device runtime information can be added to containerDevices by extending that structure to also include device runtime information. The existing Allocate function can be called from cm.PluginResourceUpdater (see my comment below) to do the actual allocation for unallocated pods.